.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,240 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: CI Workflow
+
+on:
+  push:
+    branches: [ master ]
+  pull_request_target:
+    branches:
+      - master
+
+jobs:
+  
+  build:
+    name: Build and Run Unit Tests
+    if: github.event_name == 'pull_request_target'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [12.x]
+    steps:
+    - name: Checkout Package
+      uses: actions/checkout@v2
+    - name: Using Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Clean Install
+      run: npm ci
+    - name: Build
+      run: npm run build:release
+
+  setup_integ_tests:
+    name: Check if Integ Tests are Required
+    needs: build
+    if: github.event_name == 'pull_request_target'
+    runs-on: ubuntu-latest
+    outputs:
+      integration_test_required: ${{ steps.integration_test_required.outputs.needs_integration_test }}
```

Comment:
Does this refer only to the Set integration test requirement to true ?

Reply:
WE dont need to explicitly set the output to false. We can just set it to true when it is required, and do nothing if its false. The if statement below will evaluate false if this output is not set to true.